service/textmagic/textmagic.go

```diff
@@ -0,0 +1,54 @@
+package textmagic
+
+import (
+	"context"
+	tm "github.com/textmagic/textmagic-rest-go-v2/v2"
+	"strings"
+)
+
+//TMagicService allow you to configure a TextMagic SDK client.
+type TMagicService struct {
+	UserName           string
+	APIKey             string
+	destinations       []string
+	TextMagicAPIClient *tm.APIClient
+}
+
+//NewTextMagicClient creates a new text magic client
+func NewTextMagicClient(userName, apiKey string) *TMagicService {
+
+	config := tm.NewConfiguration()
+	client := tm.NewAPIClient(config)
+
+	return &TMagicService{
+		TextMagicAPIClient: client,
+		UserName:           userName,
+		APIKey:             apiKey,
+	}
+}
+
+// AddReceivers adds the given destination phone numbers to the notifier.
+func (s *TMagicService) AddReceivers(phoneNumbers ...string) {
+	s.destinations = append(s.destinations, phoneNumbers...)
+}
+
+// Send sends a SMS via TextMagic to all previously added receivers.
+func (s *TMagicService) Send(ctx context.Context, subject, message string) error {
+
+	// put your Username and API Key from https://my.textmagic.com/online/api/rest-api/keys page.
+	auth := context.WithValue(context.Background(), tm.ContextBasicAuth, tm.BasicAuth{
+		UserName: s.UserName,
+		Password: s.APIKey,
+	})
+
+	text := subject + "\n" + message
+	_, _, err := s.TextMagicAPIClient.TextMagicApi.SendMessage(auth, tm.SendMessageInputObject{
+		Text:   text,
+		Phones: strings.Join(s.destinations, ","),
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
```suggestion
// Send sends a SMS via textmagic to all previously added receivers.
func (s *Service) Send(ctx context.Context, subject, message string) error {
	auth := context.WithValue(ctx, textMagic.ContextBasicAuth, textMagic.BasicAuth{
		UserName: s.userName,
		Password: s.apiKey,
	})

	text := subject + "\n" + message
	_, _, err := s.textMagicAPIClient.TextMagicApi.SendMessage(auth, textMagic.SendMessageInputObject{
		Text:   text,
		Phones: strings.Join(s.phoneNumbers, ","),
	})

	return err
}
```

Reply:
We're already receiving a `context.Context` from the method's arguments and thus don't need to create a new one via `context.Background()`. 

Additionally apply new field names of `Service` struct and remove unnecessary error evaluation at the end of the method.